File: week7/PartialAbstraction.cpp
#include<iostream>
using namespace std;

class Shapes
{
    private:
    virtual float area()=0;
    public:
    void displyMSG()
    {
        cout<<"This is a shape\n";
    }
};

class Triangle: public Shapes
{
    public:
    float b,h;
    Triangle(float base,float height)
    {
        b=base;
        h=height;
    }
    float area()
    {
        return 0.5*b*h;
    }
};

class Circle: public Shapes
{
    public:
    float r;
    Circle(float radius)
    {
        r=radius;
    }
    float area()
    {
        return 3.14*r*r;
    }
};

int main()
{
    Triangle tri(2,4);
    Circle cir(3.5);
    tri.displyMSG();
    cout<<"Area of the Triangle : "<<tri.area()<<endl;
    cir.displyMSG();
    cout<<"Area of the Circle: "<<cir.area()<<endl;
    return 0;
}

File: week7/PartialAbstraction.java
abstract class Shape
{
    abstract float area();
    void display()
    {
        System.out.println("This is a shape");
    }
}

class Rectangle extends Shape
{
    float a,b;
    Rectangle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return a*b;
    }
}

class Traingle extends Shape
{
    float a,b;
    Traingle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return (float)0.50*a*b;
    }
}
public class PartialAbstraction
{
    public static void main(String[] args) 
    {
        Rectangle obj=new Rectangle(23,45);
        Traingle obj1=new Traingle(67,89);
        System.out.println("Area of Rectangle : "+obj.area());
        System.out.println("Area of Triangle : "+obj1.area());
    }
}
